test/unit/liquid/forms_test.rb

```diff
@@ -114,7 +114,9 @@ def get(form, name = 'object', html_attributes={})
 
   def controller
     controller = ApplicationController.new
-    controller.stubs(:session).returns(ActionDispatch::Request.new({}).session)
+    store = stub('store', load_session: [1, {}], session_exists?: true)
+    @session = ActionDispatch::Request::Session.create(store, ActionDispatch::Request.empty, {})
+    controller.stubs(:session).returns(@session)
```

Comment:
```
  ##
  # Add a temporary stubbed method replacing +name+ for the duration
  # of the +block+. If +val_or_callable+ responds to #call, then it
  # returns the result of calling it, otherwise returns the value
  # as-is. If stubbed method yields a block, +block_args+ will be
  # passed along. Cleans up the stub at the end of the +block+. The
  # method +name+ must exist before stubbing.
  #
  #     def test_stale_eh
  #       obj_under_test = Something.new
  #       refute obj_under_test.stale?
  #
  #       Time.stub :now, Time.at(0) do
  #         assert obj_under_test.stale?
  #       end
  #     end
```
As far as I read, `#stub` replaces a method on the object called. So `stub('store', load_session: [1, {}], session_exists?: true)` should be creating a local method `#store` that you can use on line 118. IDK what `#stub` returns and why `store` variable works for this use case. But I think it is somehow confusing to use as it is now.

I see this used in signup_service_test.rb also

Reply:
I think RubyMine jumps to a wrong method definition, this method is actually from `Mocha::API`:
https://github.com/freerange/mocha/blob/v2.1.0/lib/mocha/api.rb#L77-L96

![image](https://github.com/user-attachments/assets/893b1985-b295-457a-9e8b-5c147939013a)

Reply:
So, it's just `stub` returns an object with two methods: `load_session` that returns `[1, {}]` and `session_exists?` that returns `true`

Now, I don't remember how exactly I came to the conclusion that this is exactly what those methods had to return for the session to act exactly like we need - just some digging into the code.

But what I can tell is that this fixed the following error:
```
Your application has sessions disabled. To write to the session you must first configure a session store
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/actionpack-7.0.8.6/lib/action_dispatch/request/session.rb:253:in `load_for_write!'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/actionpack-7.0.8.6/lib/action_dispatch/request/session.rb:151:in `[]='
```

And the change was introduced by this commit: https://github.com/rails/rails/commit/c1c96a014049b2660ce3a89b3c1b7aef072ae922